packages/react/src/internal/useOverflowItems.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { ReactNode, RefObject, useRef, useState } from 'react';
+import useResizeObserver from 'use-resize-observer';
+
+type Item = {
+  id: string;
+};
+
+const useOverflowItems = <T extends Item>(
```

Comment:
This would benefit from jsdoc comments/annotations so we can make sense of how to use this and what the expected behavior is. 

Just a couple examples for why
* It's not immediately clear what offsetRef is meant to be
* Is maxItems the maximum visible items or the maximum hidden items?
* When is onChange called?
* What is `itemRefHandler` and why/when would it be used?

Some examples - useControllableState, usePreviousValue, useAttachedMenu, useMatchedMedia